apps/rush-lib/src/logic/ProjectChangeAnalyzer.ts

```diff
@@ -61,14 +61,32 @@ export class ProjectChangeAnalyzer {
 
     if (this._data === undefined) {
       return undefined;
-    } else {
-      const result: Map<string, string> | undefined = this._data.get(projectName);
-      if (!result) {
-        throw new Error(`Project "${projectName}" does not exist in the current Rush configuration.`);
-      } else {
-        return result;
+    }
+
+    const project: RushConfigurationProject | undefined =
+      this._rushConfiguration.getProjectByName(projectName);
+    if (!project) {
+      throw new Error(`Project "${projectName}" does not exist in the current Rush configuration.`);
+    }
+
+    const unfilteredProjectData: Map<string, string> = this._data.get(projectName)!;
+    const filteredProjectData: Map<string, string> = new Map(unfilteredProjectData);
+
+    const ignoreMatcher: Ignore | undefined = await this._getIgnoreMatcherForProjectAsync(project, terminal);
+    if (ignoreMatcher) {
+      // At this point, `filePath` is guaranteed to start with `projectRelativeFolder`, so
+      // we can safely slice off the first N characters to get the file path relative to the
+      // root of the project.
+      for (const [filePath] of unfilteredProjectData) {
+        const relativePath: string = filePath.slice(project.projectRelativeFolder.length + 1);
+        if (ignoreMatcher.ignores(relativePath)) {
+          // Remove from the filtered data as we encounter the ignored files
+          filteredProjectData.delete(filePath);
```

Comment:
It may be faster to iterate over `unfilteredProjectData` and only populate `filteredProjectData` with the un-ignored entries than to initialize `filteredProjectData` with the entries that will end up getting removed.

Reply:
Ehhhhhhh I went back and fourth on this one. I was erring on the side of "the add/subtract operation is the expensive part and assume that there's a minimum of files ignored", but then that's also assuming initialization of a fresh map from an existing one is faster than creating a new one and adding the elements individually. Which is probably generous. I'll change it